packages/autocomplete-plugin-tags/src/__tests__/createTagsPlugin.test.tsx

```diff
@@ -0,0 +1,524 @@
+                    />
+                  </svg>
+                </button>
+              </div>
+            </div>
+          </li>,
+        ]
+      `);
+    });
+  });
+
+  test('transforms the tags source', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+
+    document.body.appendChild(panelContainer);
+
+    const tagsPlugin = createTagsPlugin({
+      initialTags: [
+        {
+          label: 'iPhone 12',
+        },
+      ],
+      transformSource({ source }) {
+        return {
+          ...source,
+          templates: {
+            item({ item }) {
+              return item.label;
+            },
+          },
+        };
+      },
+    });
+
+    autocomplete({
+      container,
+      panelContainer,
+      plugins: [tagsPlugin],
+    });
+
+    const input = container.querySelector<HTMLInputElement>('.aa-Input');
+
+    fireEvent.input(input, { target: { value: 'a' } });
+
+    await waitFor(() => {
+      expect(
+        within(
+          panelContainer.querySelector(
+            '[data-autocomplete-source-id="tagsPlugin"]'
+          )
+        ).getAllByRole('option')
+      ).toMatchInlineSnapshot(`
+        Array [
+          <li
+            aria-selected="false"
+            class="aa-Item"
+            id="autocomplete-1-item-0"
+            role="option"
+          >
+            iPhone 12
+          </li>,
+        ]
+      `);
+    });
+  });
+
+  test('fully customizes tags rendering', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+
+    document.body.appendChild(panelContainer);
+
+    const tagsPlugin = createTagsPlugin({
+      initialTags: [
+        {
+          label: 'iPhone 12',
+        },
+      ],
+      transformSource() {
+        return null;
+      },
+    });
+
+    autocomplete({
+      container,
+      panelContainer,
+      renderer: { createElement, Fragment },
+      getSources() {
+        return [
+          {
+            sourceId: 'filters',
+            getItems() {
+              return [
+                {
+                  label: 'iPhone 12',
+                },
+                {
+                  label: 'Samsung Galaxy S',
+                },
+              ];
+            },
+            templates: {
+              item({ item }) {
+                return item.label as string;
+              },
+            },
+          },
+        ];
+      },
+      plugins: [tagsPlugin],
+      render({ sections, state }, root) {
+        render(
+          <div className="aa-PanelLayout aa-Panel--scrollable">
+            <button onClick={() => state.context.tagsPlugin.setTags([])}>
```

Comment:
isn't the function called `set` / `add` / `get`, or does it get mapped somewhere I missed? 

I think there's no tests yet for `get`/`getTags`

Reply:
I think the confusion comes from the underlying [`createTags`](https://github.com/algolia/autocomplete/blob/feat/plugin-tags/plugin/packages/autocomplete-plugin-tags/src/createTags.ts) function which is used by the plugin. The function returns an [API](https://github.com/algolia/autocomplete/blob/feat/plugin-tags/plugin/packages/autocomplete-plugin-tags/src/createTags.ts) with `get`/`set`/`add`/`onChange` which the plugin then [uses internally](https://github.com/algolia/autocomplete/blob/feat/plugin-tags/plugin/packages/autocomplete-plugin-tags/src/createTagsPlugin.tsx#L81).

I've called methods like this instead of adding the suffix `*Tags` because `tags.setTags` felt a bit silly. But anyway, this is internal, so no need to test them directly.